nix-buffer.el

```diff
@@ -216,9 +216,10 @@ is removed."
   (interactive)
   (let* ((root (directory-file-name (or (buffer-file-name) default-directory)))
 	 (expr-dir (locate-dominating-file root "dir-locals.nix")))
-    (when expr-dir
-	 (let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
-	   (nix-buffer--nix-build root expr-file)))))
+    (if expr-dir
+	(let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
+	  (nix-buffer--nix-build root expr-file))
+      (error "No dir-locals.nix file found"))))
```

Comment:
Hm... I don't know if an error is appropriate here. I run `nix-buffer` in `find-file-hook`. Will loading the file fail or result in an annoying interruption if there's no dir-locals? Or maybe just a `nix-buffer-maybe` function or similar would cover what I want.